src/librustc/ty/mod.rs

```diff
@@ -592,7 +591,8 @@ impl<'tcx> serialize::UseSpecializedDecodable for &'tcx Slice<Ty<'tcx>> {}
 impl<T> Slice<T> {
     pub fn empty<'a>() -> &'a Slice<T> {
         unsafe {
-            mem::transmute(slice::from_raw_parts(0x1 as *const T, 0))
+            let slice = slice::from_raw_parts(0x1 as *const T, 0);
+            &*(slice as *const [T] as *const Slice<T>)
```

Comment:
This should be `&*(mem::align_of::<T>() as *const [T; 0] as *const [T] as *const Slice<T>)`.